eng/common/TestResources/New-TestResources.ps1

```diff
@@ -546,12 +560,28 @@ try {
             }
 
             $outputFile = "$($templateFile.originalFilePath).env"
+            if ($env:ENV_FILE) {
+                $outputFile = "$env:ENV_FILE"
+            }
 
-            $environmentText = $deploymentOutputs | ConvertTo-Json;
-            $bytes = ([System.Text.Encoding]::UTF8).GetBytes($environmentText)
-            $protectedBytes = [Security.Cryptography.ProtectedData]::Protect($bytes, $null, [Security.Cryptography.DataProtectionScope]::CurrentUser)
+            $environmentText = ''
+            if ($OutFileFormat -eq 'json') {
```

Comment:
This is going to run every time. Instead, you should default to an empty string. The `[ValidateSet]` can stay as-is because, if specified, it'll have to be one of those. But by default, an empty string means "skip all this extra processing".

Reply:
If we initialize it to empty string, then it means `-OutFileFormat` will be required when `-OutFile` is set, so I think the conditional needs to be combined with `&& $OutFile` or something.

Reply:
See https://github.com/Azure/azure-sdk-tools/pull/2053#discussion_r719718299. I think this needs to just be rolled into `$OutFile` anyway because, as @weshaggard said, we should always encrypt and `$OutFile` already was for Windows.

Reply:
How are you suggesting we determine the outfile format from the outfile switch? If we change the `-OutFile` parameter type to something like a set it's going to break people's workflows.

Reply:
I don't think we need to code the path as a parameter. See comments above. In fact, I'll just resolve this comment and we'll continue above.